libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2140,9 +2140,14 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					" outside of storage."
 				);
 		}
+
+		string functionNameSuggestion = ".";
+		if (exprType->category() == Type::Category::Function)
```

Comment:
@chriseth just changed the approach. 
I need to add extra checks to verify that the member is visible and fix other tests that are failing. 
Is this the correct approach

Reply:
Yes, this is the right approach.
Note that this PR should be tiny. No need to check visibility or computing a name suggestion.

Combine the type check here with the `memberName == "value"` check below to generate the correct second part of the error message and you are done.